src/modules/vtol_att_control/vtol_type.cpp

```diff
@@ -176,10 +177,50 @@ float VtolType::update_and_get_backtransition_pitch_sp()
 
 	const float track = atan2f(_local_pos->vy, _local_pos->vx);
 	const float accel_body_forward = cosf(track) * _local_pos->ax + sinf(track) * _local_pos->ay;
+	const float vel_body_forward = cosf(track) * _local_pos->vx + sinf(track) * _local_pos->vy;
 
-	// increase the target deceleration setpoint provided to the controller by 20%
-	// to make overshooting the transition waypoint less likely in the presence of tracking errors
-	const float dec_sp = _param_vt_b_dec_mss.get() * 1.2f;
+	const float default_dec_sp = _param_vt_b_dec_mss.get();
+	// Maximum allowed deceleration setpoint as a function of the nominal deceleration setpoint
+	const float max_dec_sp = 2.5f * default_dec_sp;
+
+	float dec_sp = default_dec_sp;
+
+	if (_attc->get_pos_sp_triplet()->current.valid && _attc->get_local_pos()->xy_valid) {
+		// Add position feedback to the deceleration setpoint calculation
+
+		// Compute backtransition end-point in local reference frame body x-direction -> dist_body_forward
+		MapProjection map_proj{_attc->get_local_pos()->ref_lat, _attc->get_local_pos()->ref_lon};
+
+		float pos_sp_x, pos_sp_y = 0.f;
+		map_proj.project(_attc->get_pos_sp_triplet()->current.lat, _attc->get_pos_sp_triplet()->current.lon, pos_sp_x,
+				 pos_sp_y);
```

Comment:
Does this mean we will not be able to back transition without a valid global position estimate? Where is this being handled?

Reply:
Hey @Jaeyoung-Lim , this is handled 7 lines above, in the `if` condition:
```
 if (_attc->get_pos_sp_triplet()->current.valid && _attc->get_local_pos()->xy_valid)
```
If the available global position is not valid, the new logic defaults to the parameter value, as before these changes :ok_hand:

Reply:
I might be wrong, but your local position setpoint can be valid, but this doesn't mean you have a valid global position estimate. Wouldn't this introduce problems with the map projection below?